Shared/Samples/Query table statistics group and sort/QueryTableStatisticsGroupAndSortView.Views.swift

```diff
@@ -0,0 +1,282 @@
+    struct StatisticRecordsList: View {
+        /// The statistic records to show in the list.
+        let records: [StatisticRecord]
+        
+        /// The field names by which to sort the records' group values.
+        let groupByFields: [String]
+        
+        /// The action to dismiss the view.
+        @Environment(\.dismiss) private var dismiss
+        
+        var body: some View {
+            List {
+                Section(groupByFields.joined(separator: ", ")) {
+                    ForEach(records, id: \.objectID) { record in
+                        if !record.statistics.isEmpty || !record.group.isEmpty {
+                            let groupDescription = record.groupValuesDescription(
+                                sortedBy: groupByFields
+                            )
+                            NavigationLink(groupDescription) {
+                                StatisticsList(statistics: record.statistics)
+                                    .navigationTitle(groupDescription)
+                            }
+                        }
+                    }
+                }
+            }
+            .navigationTitle("Statistics")
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button("Done") { dismiss() }
+                }
+            }
+        }
+    }
+}
+
+/// A list that displays given statistics.
+private struct StatisticsList: View {
+    /// The statistic names and values to show in the list.
+    let statistics: [String: any Sendable]
+    
+    var body: some View {
+        List {
+            if !statistics.isEmpty {
+                ForEach(statistics.sorted(by: { $0.key < $1.key }), id: \.key) { name, value in
+                    if let integer = value as? Int {
+                        LabeledContent(name, value: integer, format: .number)
+                    } else if let double = value as? Double {
+                        LabeledContent(name, value: double, format: .number)
+                    }
+                }
+            } else {
+                ContentUnavailableView(
+                    "No Statistics",
+                    systemImage: "list.bullet",
+                    description: Text("There are no statistics for this record.")
+                )
+            }
+        }
+    }
+}
+
+// MARK: - Extensions
+
+private extension StatisticRecord {
+    /// Returns the description of the records group value.
+    /// - Parameter groupByFields: The field names by which to sort the group
+    /// values.
+    func groupValuesDescription(sortedBy groupByFields: [String]) -> String {
+        let groupValues = groupByFields.compactMap { field in
+            if let value = group[field] {
+                "\(value)"
+            } else {
+                nil
+            }
+        }
+        
+        return groupValues.joined(separator: ", ")
```

Comment:
Use a list formatter to model best practice.

```suggestion
        return groupValues.formatted(.list(type: .and, width: .narrow))
```